tests/filecheck/dialects/complex/complex_canonicalize.mlir

```diff
@@ -0,0 +1,334 @@
+// CHECK:       %div16 = complex.constant [0x7e00 : f16, 0x7e00 : f16] : complex<f16>
+// CHECK-NEXT:  %div16_inf = complex.constant [0x7c00 : f16, 0x7c00 : f16] : complex<f16>
+// CHECK-NEXT:  %div16_minus_inf = complex.constant [0xfc00 : f16, 0xfc00 : f16] : complex<f16>
+
+%div32 = complex.div %f32_0, %f32_0 : complex<f32>
+%div32_inf = complex.div %f32_2, %f32_0 : complex<f32>
+%div32_minus_inf = complex.div %f32_1, %f32_0 : complex<f32>
+// CHECK:       %div32 = complex.constant [0x7fc00000 : f32, 0x7fc00000 : f32] : complex<f32>
+// CHECK-NEXT:  %div32_inf = complex.constant [0x7f800000 : f32, 0x7f800000 : f32] : complex<f32>
+// CHECK-NEXT:  %div32_minus_inf = complex.constant [0xff800000 : f32, 0xff800000 : f32] : complex<f32>
+
+"test.op"(%addf16, %addf16_1, %subf16, %subf16_1, %mulf16, %mulf16_1, %divf16, %divf16_1) : (complex<f16>, complex<f16>, complex<f16>, complex<f16>, complex<f16>, complex<f16>, complex<f16>, complex<f16>) -> ()
+"test.op"(%addf32, %addf32_1, %subf32, %subf32_1, %mulf32, %mulf32_1, %divf32, %divf32_1) : (complex<f32>, complex<f32>, complex<f32>, complex<f32>, complex<f32>, complex<f32>, complex<f32>, complex<f32>) -> ()
+"test.op"(%div16, %div16_inf, %div16_minus_inf) : (complex<f16>, complex<f16>, complex<f16>) -> ()
+"test.op"(%div32, %div32_inf, %div32_minus_inf) : (complex<f32>, complex<f32>, complex<f32>) -> ()
+// CHECK:       "test.op"(%addf16, %addf16_1, %subf16, %subf16_1, %mulf16, %mulf16_1, %divf16, %divf16_1) : (complex<f16>, complex<f16>, complex<f16>, complex<f16>, complex<f16>, complex<f16>, complex<f16>, complex<f16>) -> ()
+// CHECK-NEXT:  "test.op"(%addf32, %addf32_1, %subf32, %subf32_1, %mulf32, %mulf32_1, %divf32, %divf32_1) : (complex<f32>, complex<f32>, complex<f32>, complex<f32>, complex<f32>, complex<f32>, complex<f32>, complex<f32>) -> ()
+// CHECK-NEXT:  "test.op"(%div16, %div16_inf, %div16_minus_inf) : (complex<f16>, complex<f16>, complex<f16>) -> ()
+// CHECK-NEXT:  "test.op"(%div32, %div32_inf, %div32_minus_inf) : (complex<f32>, complex<f32>, complex<f32>) -> ()
+
+
+// CHECK-LABEL: func @test_create_of_real_and_imag
+// CHECK-SAME: (%[[CPLX:.*]] : complex<f32>) -> complex<f32>
+func.func @test_create_of_real_and_imag(%cplx: complex<f32>) -> complex<f32> {
+  %real = complex.re %cplx : complex<f32>
+  %imag = complex.im %cplx : complex<f32>
+  %complex = complex.create %real, %imag : complex<f32>
+  return %complex : complex<f32>
+
+  // CHECK:  return %[[CPLX]] : complex<f32>
+}
+
+
+// CHECK-LABEL: func @test_create_of_real_and_imag2
+func.func @test_create_of_real_and_imag2() -> complex<f32> {
+  %cplx = "test.op"() : () -> (complex<f32>)
+  %real = complex.re %cplx : complex<f32>
+  %imag = complex.im %cplx : complex<f32>
+  %complex = complex.create %real, %imag : complex<f32>
+  return %complex : complex<f32>
+
+  // CHECK:       %[[CPLX:.*]] = "test.op"() : () -> complex<f32>
+  // CHECK-NEXT:  return %[[CPLX]] : complex<f32>
+}
```

Comment:
what's the benefit of this test vs the one above? feels like we can get rid of it